lib/apiservers/engine/backends/container.go

```diff
@@ -1032,11 +1053,18 @@ func (c *Container) ContainerTop(name string, psArgs string) (*types.ContainerPr
 
 // Containers returns the list of containers to show given the user's filtering.
 func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Container, error) {
+	defer trace.End(trace.Begin(fmt.Sprintf("ListOptions %#v", config)))
```

Comment:
@cgtexmex This function is pretty big and complex.  Can we break it up into some util functions?